src/trunk.c

```diff
@@ -1170,31 +1201,45 @@ trunk_set_pivot(trunk_handle *spl,
 static inline void
 trunk_set_initial_pivots(trunk_handle *spl,
                          page_handle  *node,
-                         const char   *min_key,
-                         const char   *max_key)
+                         key           min_key,
+                         key           max_key)
 {
    debug_assert(trunk_key_compare(spl, min_key, max_key) < 0);
 
    trunk_hdr *hdr      = (trunk_hdr *)node->data;
    hdr->num_pivot_keys = 2;
 
-   char *dst_pivot_key = trunk_get_pivot(spl, node, 0);
-   memmove(dst_pivot_key, min_key, trunk_key_size(spl));
-   trunk_pivot_data *pdata = trunk_get_pivot_data(spl, node, 0);
+   char             *dst_pivot_key = trunk_get_pivot_buffer(spl, node, 0);
+   trunk_pivot_data *pdata         = trunk_get_pivot_data(spl, node, 0);
    ZERO_CONTENTS(pdata);
    pdata->srq_idx = -1;
-   dst_pivot_key  = trunk_get_pivot(spl, node, 1);
-   memmove(dst_pivot_key, max_key, trunk_key_size(spl));
+   if (key_is_negative_infinity(min_key)) {
```

Comment:
That would be a hassle because `trunk_set_pivot` checks the sortedness of the pivots in debug mode, which would fail when setting the first pivot (since the second hasn't been set).

To avoid this, we'd have to set `hdr->num_pivot_keys` to 1, then set the first pivot, then set the `num_pivot_keys` to 2, then set the second pivot.  Seems like unnecessary hassle to me.